.gitattributes

```diff
@@ -1,5 +1,6 @@
 /src/test/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/InputNewlineAtEndOfFileLf.java eol=lf
 /src/test/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/InputNewlineAtEndOfFileCrlf.java eol=crlf
 /src/test/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/InputNewlineAtEndOfFileCr.java -text
+/src/test/resources/com/puppycrawl/tools/checkstyle/checks/newlineatendoffile/InputNewlineAtEndOfFileNewlineAtEnd.txt eol=lf
```

Comment:
Please don't convert an existing file to only be a specific line ending.
Either use an existing one or create a new one and put `Lf` on the end of the file name like the others to denote its ending.

Reply:
Ok, I renamed it, as like mentioned before, the other test using it also needs it like that or will break as soon as #4074 gets fixed.

Reply:
> Ok, I renamed it

Like I said, don't convert an existing file. It is probably testing something specific since appveyor will automatically convert files not mentioned in `.gitattributes` to `\r\n`.
Please just make a whole new file.

Reply:
I'm sorry, but I don't get it.
The test tests, that the check works properly if the file only consists of an empty line with the line separator option set to LF.
If #4074 gets fixed, this test will still be green on systems where the files get checked out with LF but fail on systems where the files get checked out with CRLF endings.
Updating the expected result doesn't sound like the correct way to me, it intermixes two tests, the test that the check works properly on a file consisting only of an empty line and that LF does not match CRLF.
Imho it is better to test the one aspect in the test that it is about and have a different test that tests the bug that #4074 is about.

Reply:
> If #4074 gets fixed, this test will still be green

Ok, I thought you were saying this test would fail once another issue is worked on. I misunderstood what you meant.

We still need the 2 different input files, one goes to the git's default and one that is forced to LF.

Reply:
> Ok, I thought you were saying this test would fail once another issue is worked on. I misunderstood what you meant.

No you didn't. o_O
That is **exactly** what I am saying.
The test we are talking about verifies that the check works properly if the parameter is set to `LF` and the file consists solely of `LF`.
Currently this works also if the file consists of `CR` followed by `LF` due to the mentioned issue.
If the mentioned issue gets fixed, then this test will fail on checkouts that use CRLF line endings.
By using the same file here, this will not happen.

> We still need the 2 different input files, one goes to the git's default and one that is forced to LF.

I can surely do this, I just want you to first understand what I am saying, as you obviously didn't yet, so that you can decide with the correct facts in mind that I'm stating. :-)

Reply:
> this test will still be green on systems where the files get checked out with LF but fail on systems where the files get checked out with CRLF endings.

We will have to make the expected result conditional if we go forward with different line endings on the same OS.

> Imho it is better to test the one aspect in the test that it is about and have a different test that tests the bug that #4074 is about.

Thats what we are doing isn't it? We are adding a new file that is LF only and 4074 will add a new file that is CRLF only (if there isn't one). All I am saying is there is a 3rd file which is basically AUTO line endings and it shouldn't be touched in the scope of this PR. It is testing something else and should remain as it is.

> I just want you to first understand what I am saying, as you obviously didn't yet, so that you can decide with the correct facts in mind that I'm stating.

I am not sure we are on the same page with this long discussion. Lets move on and not slow down the PR.
We should create a new file and not rename the old one.

Reply:
> > Imho it is better to test the one aspect in the test that it is about and have a different test that tests the bug that #4074 is about.
> 
> Thats what we are doing isn't it? We are adding a new file that is LF only and 4074 will add a new file that is CRLF only (if there isn't one). All I am saying is there is a 3rd file which is basically AUTO line endings and it shouldn't be touched in the scope of this PR. It is testing something else and should remain as it is.

I don't think so.
Despite its name, the test `testFileWithEmptyLineOnly` tests how the check behaves with `LineSeparatorOption.LF` set and a file that only consists of an empty line.
If you currently check out on a system with `CR` line endings, this test already fails, as it is a bug that the test is green with `CRLF ` line endings (which will be fixed with #4074).
The test imho is not there to test that the check behaves properly with different line endings, or the test should use files with different line endings to verify it works properly with them.

> We should create a new file and not rename the old one.

Well, as you wish, I pushed an update.

Reply:
> the test should use files with different line endings to verify it works properly with them

There should be a separate test for each kind of line ending. So in the end we should have atleast 20 tests ( 4 file line endings LF, CRLF, CR, system x 5 options crlf, cr, lf, lf_cr_crlf, system).
`system` will always be the hardest to test since it can pass on 1 system and fail on another.

> out on a system with CR line endings, this test already fails

MacOS is CR line endings right? We already have MacOS in the CI and it is passing. Why isn't it failing if that is the case?

Reply:
OS X nowadays is LF too.
Older versions were CR.